default/AI/ReadChokePoints.py

```diff
@@ -0,0 +1,124 @@
+    used_nodes = set()
+    node_values.clear()
+    shortest_paths = nx.all_pairs_shortest_path(G)
+    for source in shortest_paths:
+        used_nodes.add(source)
+        for target in shortest_paths[source]:
+            if target in used_nodes:
+                continue
+            this_path = shortest_paths[source][target]
+            if not any(node in owned_nodes for node in this_path):
+                weight = 1e-12  # assign a small weight here to prefer chokepoints
+            else:
+                weight = 0
+                if any(node in enemy_nodes for node in this_path):
+                    weight += 20
+                if any(node in unexplored_nodes for node in this_path):
+                    weight += 10
+            idx_owned = [index for index, node in enumerate(this_path) if node in owned_nodes]
+            idx_threat = [index for index, node in enumerate(this_path) if node in enemy_nodes or node in unexplored_nodes]
+            for this_index, node in enumerate(this_path):
+                distance_to_owned = min(abs(this_index - idx) for idx in idx_owned) if idx_owned else 0
+                distance_to_threat = min(abs(this_index - idx) for idx in idx_threat) if idx_threat else 0
+                aggression_factor = 1  # a measure on how "aggressive" (how close to enemy systems) the found points should be
+                distance_factor = max(distance_to_owned, aggression_factor*distance_to_threat, 1)
+                node_values[node] = node_values.get(node, 0) + weight / distance_factor
+    for node in node_values:
+        if node in unexplored_nodes:
+            node_values[node] = 0
+        else:
+            if node in owned_nodes:
+                node_values[node] *= 1.2  # prefer owned systems as planets provide additional defensive resources
+        if node in all_defensive_positions:
+            node_values[node] = 0  # count each node only once.
+    maxval = max(node_values.iteritems(), key=operator.itemgetter(1))[1]
+    return True if maxval > 1 else False  # to end while loop if all threats are separated from our planets
+
+
+def draw(G):
+    edges = [(u, v) for (u, v) in G.edges()]
+    pos = {n: data['position'] for n, data in G.nodes(data=True)}  # positions for all nodes
+    nx.draw_networkx_nodes(G, pos, node_size=100,
+                           node_color=[data['color'] for n, data in G.nodes(data=True)],
+                           )
+    # edges
+    nx.draw_networkx_edges(G, pos, edgelist=edges, width=1, alpha=0.5, edge_color='b', style='dashed')
+    # labels
+    pos = {k: (a, b - 15) for k, (a, b) in pos.items()}
+    nx.draw_networkx_labels(G, pos, font_size=10, font_family='Arial', labels={n: data['name'] for n, data in G.nodes(data=True)})
+    plt.axis('off')
+    # plt.savefig("universe.png")  # save as png
+    plt.show(block=True)
+
+
+# read file
+empireID = 2
+filename = r'C:\Program Files (x86)\FreeOrion\ KnownUniverseAI%d.pkl' % empireID
```

Comment:
:) 

Do you run game as admin or just use virtual store?   

You can pass file name via console and get it sys.argv[1]